### ISSUE TITLE
Enhance documentation files for IDs associated with multiple rules

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn-analyzers#4808]
See #4798, which adds a new rule with an existing id.

The documentation files only have one rule as shown in diff:

![image](https://user-images.githubusercontent.com/31348972/107388395-7ad5c700-6afe-11eb-8615-e11370a0c5a4.png)

FYI @sharwell

**Comment by sharwell** [comment]
> which adds a new rule with an existing id

It's still the same rule. The part that changed is adding a description for why it applies in a specific situation. The documentation file was also incorrect before this change.

**Comment by Youssef1313** [comment]
@sharwell Yeah I see now it was already incorrect. But aren't the following considered different rules (which are existing prior your PR):

https://github.com/sharwell/roslyn-analyzers/blob/14703b1a232f71b79b3a9c895b9d14747eba0173/src/Roslyn.Diagnostics.Analyzers/Core/AbstractDoNotCopyValue.cs#L50-L70

Or that's not the correct terminology?

**Comment by mavasani** [comment]
We need one of the two things:
1. Ability to pick a descriptor for documentation files
2. Proposal to show information from all descriptors
